src/test/kotlin/uk/gov/justice/hmpps/offendersearch/services/MatchServiceTest.kt

```diff
@@ -357,17 +571,25 @@ internal class MatchServiceTest {
 
 }
 
+private fun String.toLocalDate(): LocalDate = LocalDate.parse(this)
+
 fun SearchRequest.mustNames(): Map<String, Any> {
   val query = source().query() as BoolQueryBuilder
   return query.must().filterIsInstance<MatchQueryBuilder>().map { it.fieldName() to it.value() }.toMap()
 }
 
 fun SearchRequest.nestedShouldMultiMatchNames(): Map<Any, List<String>> {
   val query = source().query() as BoolQueryBuilder
-  return query.must().filterIsInstance<BoolQueryBuilder>().flatMap { it.should().filterIsInstance<MultiMatchQueryBuilder>() }.map { it.value() to it.fields().keys.toList().sorted()  }.toMap()
+  return query.must().filterIsInstance<BoolQueryBuilder>().flatMap { it.should().filterIsInstance<MultiMatchQueryBuilder>() }.map { it.value() to it.fields().keys.toList().sorted() }.toMap()
 }
 
 fun SearchRequest.mustMultiMatchNames(): Map<Any, List<String>> {
   val query = source().query() as BoolQueryBuilder
   return query.must().filterIsInstance<MultiMatchQueryBuilder>().map { it.value() to it.fields().keys.toList().sorted() }.toMap()
 }
+
+fun QueryBuilder.mustNames(): Map<String, Any> {
```

Comment:
Just wondering why you extended the interface then cast to an instance?

Reply:
the caller only knows about the interface (although I know in the context it is called it will be the correct subclass) so I had to but the cast somewhere, maybe I should convert to a `when` to be a bit clear rather than a cast. Thanks